presto-spi/src/main/java/com/facebook/presto/spi/block/SliceArrayBlock.java

```diff
@@ -201,25 +208,31 @@ public int getRegionSizeInBytes(int positionOffset, int length)
             throw new IndexOutOfBoundsException("Invalid position " + positionOffset + " in block with " + positionCount + " positions");
         }
 
-        long sizeInBytes = length * (long) (ARRAY_OBJECT_INDEX_SCALE + SLICE_INSTANCE_SIZE);
-        for (int i = positionOffset; i < positionOffset + length; i++) {
-            Slice value = values[i];
-            if (value != null) {
-                sizeInBytes += value.length();
-            }
-        }
-        return intSaturatedCast(sizeInBytes);
+        return getSliceArraySizeInBytes(values, positionOffset, length);
     }
 
     static int getSliceArrayRetainedSizeInBytes(Slice[] values)
     {
-        long sizeInBytes = SizeOf.sizeOf(values);
+        long sizeInBytes = sizeOf(values);
         Map<Object, Boolean> uniqueRetained = new IdentityHashMap<>(values.length);
+        boolean emptySliceExists = false;
         for (Slice value : values) {
-            if (value != null && value.getBase() != null && uniqueRetained.put(value.getBase(), true) == null) {
-                sizeInBytes += value.getRetainedSize();
+            if (value != null) {
+                if (value.getBase() != null) {
+                    if (uniqueRetained.put(value.getBase(), true) == null) {
+                        sizeInBytes += value.getRetainedSize();
+                    }
+                }
+                else if (value.equals(EMPTY_SLICE)) {
```

Comment:
`base` can be null for empty slices and for other cases, so to be safe I only took care of the empty slice case (which the test exercises).